Convert second 1/2 x-pack plugins from integTest to [yaml | java]RestTest or internalClusterTest

jakelandis opened the pull request:
For 1/2 the plugins in x-pack, the integTest
task is now a no-op and all of the tests are now executed via a test,
yamlRestTest, javaRestTest, or internalClusterTest.

This includes the following projects:
security, spatial, stack, transform, vecotrs, voting-only-node, and watcher.

A few of the more specialized qa projects within these plugins
have not been changed with this PR due to additional complexity which should
be addressed separately. 

related: #60630
related: #56841
related: #59939
related: #55896

jakelandis committed:
most of security

jakelandis committed:
security internalCluster

jakelandis committed:
finish up security

jakelandis committed:
security:qa

jakelandis committed:
spatial

jakelandis committed:
stack

jakelandis committed:
transform

jakelandis committed:
vectors

jakelandis committed:
voting-only-node

jakelandis committed:
watcher - internalClusterTest

jakelandis committed:
watcher qa

jakelandis committed:
Merge remote-tracking branch 'upstream/master' into yamlRestTest_xpack_plugins_part2

jakelandis committed:
Merge branch 'master' of github.com:elastic/elasticsearch into yamlRestTest_xpack_plugins_part2

jakelandis committed:
remove commented out line

jakelandis committed:
spotless

jakelandis committed:
fix precommit

jakelandis commented:
@elasticmachine run elasticsearch-ci/packaging-sample-windows

elasticmachine commented:
Pinging @elastic/es-core-infra (:Core/Infra/Build)

jakelandis commented on x-pack/plugin/watcher/qa/common/build.gradle:
```diff
@@ -0,0 +1,20 @@
+apply plugin: 'elasticsearch.build'
```
I introduced a qa/common project to better match how the other projects have this configured.

mark-vieira replied:
Fair enough, but I don't see the point in the additional `testJar`. We're literally just defining another jar that's identical to the default jar. Why another artifact? Can't we just depend on this project via the default configuraiton?

jakelandis replied:
thanks... fixed on https://github.com/elastic/elasticsearch/pull/61802/commits/667ea2abd20fc50c91e8ac16f5066ed44aad0d9d

jakelandis commented on x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/actions/email/EmailActionTests.java:
```diff
@@ -605,4 +607,18 @@ private EmailAttachments randomEmailAttachments() throws IOException {
         return new EmailAttachments(attachments);
     }
+
+    public static class NoopEmailService extends EmailService {
```
minor duplication here to avoid sharing classpath's between test and javaRestTests

mark-vieira replied:
Could this go in the "common" qa project to be shared then?

jakelandis replied:
i would prefer not to have the unit tests depend on the qa project (which has the parent classes for the java and yaml rest test runners). creating a dependency the other way requires a new test artifact or test fixture (in this case I prefer the minor duplication)

mark-vieira replied:
No worries. It's a tiny implementation so it's probably not a big deal for a bit of duplication here.

jakelandis commented on x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/transport/filter/IpFilteringIntegrationTests.java:
```diff
@@ -3,7 +3,7 @@
  * you may not use this file except in compliance with the Elastic License.
  */
-package org.elasticsearch.xpack.security.transport.filter;
+package org.elasticsearch.transport.filter;
```
didn't mean to change the package ... fixing now ..

jakelandis committed:
fix package location

jakelandis committed:
fix package location

mark-vieira commented on x-pack/plugin/security/build.gradle:
```diff
@@ -476,6 +484,11 @@ test {
   systemProperty 'es.transport.buffer.size', '256k'
 }
 
+internalClusterTest {
```
I suspect this probably isn't needed for unit tests. Can we ditch the block above?

jakelandis replied:
yup: https://github.com/elastic/elasticsearch/pull/61802/commits/1f4ca9a0ea0276a9c23b9ec69b4ad0e1f6a01bf7

mark-vieira commented on x-pack/plugin/watcher/build.gradle:
```diff
@@ -86,6 +87,10 @@ test {
 }
 
+internalClusterTest {
+  systemProperty 'es.set.netty.runtime.available.processors', 'false'
```
Same, I suspect this is _only_ needed for integration tests.

jakelandis replied:
indeed: https://github.com/elastic/elasticsearch/pull/61802/commits/1f4ca9a0ea0276a9c23b9ec69b4ad0e1f6a01bf7

mark-vieira reviewed (approved):
Left a few comments but otherwise LGTM.

jakelandis committed:
remove unecessary watcher test artifact

jakelandis commented on x-pack/plugin/security/qa/basic-enable-security/build.gradle:
```diff
@@ -1,36 +1,36 @@
+task javaRestTestWithSecurity(type: StandaloneRestIntegTestTask) {
+  description = "Run tests against a cluster that has security enabled"
+  useCluster testClusters.javaRestTest
+  dependsOn javaRestTest
```
yeah... that tripped me up at first sight too.  This is testing that enabling security where it was not enabled prior works. e.g. start without security run some tests, enable security restart then run the same tests. There is a boolean read by the tests that controls what to assert with/without security enabled.

jakelandis committed:
remove unnecessary config

jakelandis commented:
thanks @mark-vieira - I addressed your comments. 

Also, I held off merging the first half https://github.com/elastic/elasticsearch/pull/60630 since I didn't want to merge 1/2 a major refactoring then go on vacation :) ... so I will merge these two PRs back to back.
